packages/notification-service/src/blockscout/transfers.ts

```diff
@@ -53,52 +53,44 @@ async function getLatestTokenTransfers(
   }
 
   console.debug('New logs found for token:', tokenAddress, response.result.length)
-  const { transfers, latestBlock } = useLogs
-    ? decodeLogs(response.result as Log[])
-    : formatNativeTransfers(response.result as TokenTransfer[])
-  for (const txTransfers of transfers.values()) {
-    txTransfers.forEach((t) => (t.currency = currency))
-  }
+  const { transfers, latestBlock } = formatTransfers(response.result as TokenTransfer[], currency)
   return { transfers, latestBlock }
 }
 
+function toTransferArray(transfersByTx: Map<string, Transfer[]>) {
+  return [...transfersByTx?.values()]
+    .filter((transferList) => transferList.length > 0)
+    .map((transferList) => {
+      const amount = Math.max(...transferList.map((t) => parseFloat(t.value)))
+      const transfer = transferList.find((t) => parseFloat(t.value) === amount)
+      return transfer!
+    })
+}
```

Comment:
My understanding was that each tx id was for a single one, but they are coming repeated from Blockscout so I did this to avoid sending more than one push notification for the same tx.
There might be something I'm misunderstanding though, I'll dive a little deeper to see if there's something else here that could be used to avoid sending duplicates.

Reply:
Ok, finally understood this better. It was a problem with blockscout that returned duplicate transfers for the same tx hash. Here's the PR that fixes it: https://github.com/celo-org/blockscout/pull/208

I removed getting the max thing now and everything's looking good :) Just as a note this can't be delpoyed to mainnet yet since the fix is only on Alfajores so far.